src/service/search/datafusion/arrzip_udf.rs

```diff
@@ -0,0 +1,231 @@
+                // in arrow, any value can be null.
+                // Here we decide to make our UDF to return null when either argument is null.
+                (Some(arr_field1), (Some(arr_field2), Some(delim))) => {
+                    let arr_field1: json::Value =
+                        json::from_str(arr_field1).expect("Failed to deserialize arrzip field1");
+                    let arr_field2: json::Value =
+                        json::from_str(arr_field2).expect("Failed to deserialize arrzip field2");
+                    let mut zipped_arrs = vec![];
+                    if let (json::Value::Array(field1), json::Value::Array(field2)) =
+                        (arr_field1, arr_field2)
+                    {
+                        // Field1 and field2 can be of different types
+                        zip(field1.iter(), field2.iter()).for_each(|(field1, field2)| {
+                            let field1 = if field1.is_boolean() {
+                                let field1 = field1.as_bool().unwrap();
+                                field1.to_string()
+                            } else if field1.is_f64() {
+                                let field1 = field1.as_f64().unwrap();
+                                field1.to_string()
+                            } else if field1.is_i64() {
+                                let field1 = field1.as_i64().unwrap();
+                                field1.to_string()
+                            } else if field1.is_u64() {
+                                let field1 = field1.as_u64().unwrap();
+                                field1.to_string()
+                            } else if field1.is_string() {
+                                let field1 = field1.as_str().unwrap();
+                                field1.to_string()
+                            } else {
+                                "".to_string()
+                            };
+                            let field2 = if field2.is_boolean() {
+                                let field2 = field2.as_bool().unwrap();
+                                field2.to_string()
+                            } else if field2.is_f64() {
+                                let field2 = field2.as_f64().unwrap();
+                                field2.to_string()
+                            } else if field2.is_i64() {
+                                let field2 = field2.as_i64().unwrap();
+                                field2.to_string()
+                            } else if field2.is_u64() {
+                                let field2 = field2.as_u64().unwrap();
+                                field2.to_string()
+                            } else if field2.is_string() {
+                                let field2 = field2.as_str().unwrap();
+                                field2.to_string()
+                            } else {
+                                "".to_string()
+                            };
```

Comment:
Make it rather parametrized for e.g. a function which can test and do it for you, to avoid repetition.